tests/commons/j2/test_j2_functions.py

```
@@ -11,290 +12,323 @@
+    assert "c" in map_fmt._map, "key not added to map"
+    assert len(map_fmt._map) == len(map_expects), "key not added correctly"
+    assert map_fmt._map["c"] == map_expects["c"], "value not added correctly"
 
-    def test_add(self):
-        key_name = "Alpha-5 code"
-        key_val = 9
-        self.fmt.add(key_name, key_val)
-        self.assertDictEqual(
-            self.fmt._map, self.fmt._map | {key_name: key_val}
+    # check applying the map
+    for origin in map_expects.keys():
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)